zarr/convenience.py

```diff
@@ -1179,4 +1179,5 @@ def open_consolidated(store, metadata_key='.zmetadata', mode='r+', **kwargs):
     meta_store = ConsolidatedMetadataStore(store, metadata_key=metadata_key)
 
     # pass through
-    return open(store=meta_store, chunk_store=store, mode=mode, **kwargs)
+    chunk_store = kwargs.pop('chunk_store', None) or store
+    return open(store=meta_store, chunk_store=chunk_store, mode=mode, **kwargs)
```

Comment:
```suggestion
    return open(store=meta_store, mode=mode, **kwargs)
```

The `chunk_store` argument can just be treated as one of the `kwargs`, it will get handled downstream.

Reply:
One test fails with the above change:
```
=================================== FAILURES ===================================
__________________________ test_consolidate_metadata ___________________________

    def test_consolidate_metadata():
    
        # setup initial data
        store = MemoryStore()
        z = group(store)
        z.create_group('g1')
        g2 = z.create_group('g2')
        g2.attrs['hello'] = 'world'
        arr = g2.create_dataset('arr', shape=(20, 20), chunks=(5, 5), dtype='f8')
        assert 16 == arr.nchunks
        assert 0 == arr.nchunks_initialized
        arr.attrs['data'] = 1
        arr[:] = 1.0
        assert 16 == arr.nchunks_initialized
    
        # perform consolidation
        out = consolidate_metadata(store)
        assert isinstance(out, Group)
        assert '.zmetadata' in store
        for key in ['.zgroup',
                    'g1/.zgroup',
                    'g2/.zgroup',
                    'g2/.zattrs',
                    'g2/arr/.zarray',
                    'g2/arr/.zattrs']:
            del store[key]
    
        # open consolidated
        z2 = open_consolidated(store)
        assert ['g1', 'g2'] == list(z2)
        assert 'world' == z2.g2.attrs['hello']
        assert 1 == z2.g2.arr.attrs['data']
>       assert (z2.g2.arr[:] == 1.0).all()
E       assert False
E        +  where False = <built-in method all of numpy.ndarray object at 0x1099bb8f0>()
E        +    where <built-in method all of numpy.ndarray object at 0x1099bb8f0> = array([[0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n        0., 0., 0., 0.],\n       [0., 0., 0., 0...     0., 0., 0., 0.],\n       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,\n        0., 0., 0., 0.]]) == 1.0.all

zarr/tests/test_convenience.py:126: AssertionError
```

Reply:
I was being dim and didn't look at this properly. My suggestion is wrong, please ignore. Sorry for wasting time.